### PR TITLE
[Merged by Bors] - feat: `Matrix.trace_units_conj`

### PR DESCRIPTION
**Pull request by eric-wieser**
This matches `Matrix.det_units_conj` and friends.

This doesn't actually help for the case of invertible matrices indexed by different indices, but is handy for square ones.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/LinearAlgebra/Matrix/NonsingularInverse.lean [hunk 1]
```diff
@@ -722,4 +722,21 @@ theorem det_conj' {M : Matrix m m α} (h : IsUnit M) (N : Matrix m m α) :
...
+/-! ### More results about traces -/
+
+
+section trace
+
+variable [Fintype m] [DecidableEq m]
+
+/-- A variant of `Matrix.trace_units_conj`. -/
+theorem trace_conj {M : Matrix m m α} (h : IsUnit M) (N : Matrix m m α) :
+    trace (M * N * M⁻¹) = trace N := by rw [← h.unit_spec, ← coe_units_inv, trace_units_conj]
```
[the remark this answers is not in the document]
Comment (j-loreaux) [reply Mathlib/LinearAlgebra/Matrix/NonsingularInverse.lean:734]: I didn't even know we had `IsConj`!

[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/LinearAlgebra/Matrix/NonsingularInverse.lean:734]: My preference would be to postpone that to a later refactor; writing `conj` lemmas the way I do in this PR is reasonably established, and `IsConj` is probably only beneficial if we apply it everywhere.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**